apps/server/src/modules/media-source-sync/service/media-source-sync.service.ts

```diff
@@ -35,6 +38,36 @@ export class MediaSourceSyncService {
 		return report;
 	}
 
+	public async fetchMediumMetadata(
+		mediumId: string,
+		mediaSourceId: string | undefined,
+		mediaSourceDataFormat: MediaSourceDataFormat | undefined
+	): Promise<MediaMetadataDto> {
+		if (!mediaSourceDataFormat) {
+			throw new MediaSourceDataFormatMissingLoggableException();
+		}
+
+		if (!mediaSourceId) {
+			throw new MediaSourceIdMissingLoggableException(mediaSourceDataFormat);
+		}
```

Comment:
the format can be read from the media source and the source id should never be undefined at this point